From 7b23b89961af10ccc748d90cdef6b516b04a5d3b Mon Sep 17 00:00:00 2001
From: Wei-Chiu Chuang <weichiu@cloudera.com>
Date: Thu, 25 Feb 2016 11:23:32 -0800
Subject: [PATCH 1273/1380] CLOUDERA-BUILD. CDH-32824. S3a properties
 deprecation warning is logged when not necessary

Change-Id: I016dc71b7e9a3ac7e2ae42cff0d08b09326997f8
(cherry picked from commit b6381d50eb5e340f2a7ecbf2f9f1c2532db96473)
(cherry picked from commit fc7edb7b3aa79983e5f21c92b2b4b8a8cb65d432)
---
 .../org/apache/hadoop/fs/s3a/S3AFileSystem.java    |   12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java b/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java
index 45ba062..eb6ad77 100644
--- a/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java
+++ b/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java
@@ -172,10 +172,14 @@ public void initialize(URI name, Configuration conf) throws IOException {
 
     // Try to get our credentials or just connect anonymously
     // CLOUDERA-BUILD: deprecated keys are alias of supported keys.
-    String accessKey = conf.get(
-        ACCESS_KEY, conf.get(DEPRECATED_ACCESS_KEY, null));
-    String secretKey = conf.get(
-        SECRET_KEY, conf.get(DEPRECATED_SECRET_KEY, null));
+    String accessKey = conf.get(ACCESS_KEY, null);
+    if (accessKey == null) {
+      accessKey = conf.get(DEPRECATED_ACCESS_KEY, null);
+    }
+    String secretKey = conf.get(SECRET_KEY, null);
+    if (secretKey == null) {
+      secretKey = conf.get(DEPRECATED_SECRET_KEY, null);
+    }
 
     String userInfo = name.getUserInfo();
     if (userInfo != null) {
-- 
1.7.9.5

